### PR TITLE
Add AdaNorm support through MLP, transformer, and ViT

### PR DESCRIPTION
**Pull request by TidalPaladin**
## Summary
- add standalone adaptive normalization primitives with a compiled `ada_norm(...)` functional and `AdaNorm` module
- extend the fused `NormMLP` path with optional norm modulation and output gating, then add `AdaNormMLP` for AdaLN-Zero style conditioned MLPs
- propagate opt-in conditioning through transformer layers and `ViT` so a single conditioning vector can modulate every block MLP
- add configurable `adaln_gate_init` so conditioned models can either keep AdaLN-Zero defaults (`0.0`) or preserve pretrained unconditioned MLP behavior (`1.0`) during conversion

## Changes
- add adaptive norm modulation support to `vit/norm.py` and `vit/fused.py`
- add `conditioning_size` support to transformer encoder, decoder, and cross-attention layers in `vit/transformer.py`
- add `conditioning_size`, `adaln_gate_init`, and `conditioning=` plumbing to `ViTConfig` and `ViT` in `vit/vit.py`
- validate conditioned vs unconditioned usage explicitly so configured models require `conditioning` and unconfigured models reject it
- handle selective stochastic depth and activation checkpointing with conditioned MLP calls
- avoid double-zeroing conditioned transformer MLP residual outputs so AdaNormMLP paths train correctly at initialization
- add regression coverage for standalone AdaNorm, AdaNormMLP, conditioned transformer layers, conditioned ViT forwards, conditioned attention-weight extraction, checkpointing, and pretrained-compatible gate init

## Test Plan
- [x] `make check`
- [x] `uv run pytest tests/test_norm.py tests/test_fused.py`
- [x] `uv run pytest tests/test_transformer.py tests/test_vit.py tests/test_checkpointing.py`

## Notes
- conditioning is opt-in via `conditioning_size`; existing unconditioned model paths keep their prior behavior
- `adaln_gate_init=0.0` keeps the default AdaLN-Zero initialization for scratch training
- `adaln_gate_init=1.0` is intended for converting pretrained unconditioned MLP stacks into conditioned ones while preserving the loaded MLP path at initialization

### CODE REVIEW

File: vit/fused.py [hunk 1]
```diff
@@ -249,10 +285,104 @@ def forward(self, x: Tensor) -> Tensor:
...
+        factory_kwargs = {"device": device, "dtype": dtype}
+        self.conditioning_size = hidden_size if conditioning_size is None else conditioning_size
+        self.conditioning_activation = nn.SiLU()
+        self.modulation = nn.Linear(self.conditioning_size, 3 * hidden_size, **factory_kwargs)
+        self.reset_modulation_parameters()
+
+    @torch.no_grad()
+    def reset_modulation_parameters(self) -> None:
+        nn.init.zeros_(self.modulation.weight)
+        zero_bias_if_present(self.modulation)
```
[the remark this answers is not in the document]
Comment (TidalPaladin) [reply vit/fused.py:358]: Addressed in `6692644`.

I kept the AdaNormMLP gate at its AdaLN-zero default and fixed the freeze by removing the second zero-init in transformer construction: encoder/decoder/cross-attention blocks now skip zeroing `mlp.fc2` when the MLP is `AdaNormMLP`, while plain `NormMLP` still keeps the existing zero-residual init. That preserves zero-output conditioned residuals at init, but the gate slice now gets nonzero gradients immediately instead of being dead.

I also removed the test-only gate-bias overrides and added a regression that checks the conditioned transformer gate bias receives nonzero gradient at initialization.

Validation:
- `make check`
- `uv run pytest tests/test_fused.py tests/test_transformer.py tests/test_vit.py tests/test_checkpointing.py`
